### PR TITLE
Finalize unification or normalization of RPC calls

### LINKED ISSUE #14: Unification of RPC labels, commands, descriptions
As continuation of https://github.com/OmniLayer/omnicore/pull/3#issuecomment-90286069:

> @zathras-crypto: I would say for OmniCore at least the API documentation should be the primary doc? It'll need updating if my work gets merged in that's for sure :)

> https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/doc/apidocumentation.md

> I would say for OmniCore at least the API documentation should be the primary doc?

Well, this is the documentation, but I was more thinking about something more compact, and with the intention of finding a naming convention for various places, such as Omni Core, Omni Wallet, OmniJ, ...

Where I was basically going: there are several labels and commands, documentation pieces and more, all referring to the same thing:

- Spec: `"Create a Property via Crowdsale with Variable number of Tokens"`
- Omni Core RPC command: `"sendissuancecrowdsale_OMNI"`
- Omni Core RPC description field: `"Create Property - Variable"`
- Omni Core transaction history: `"Create Property"`
- Omni Wallet short description: `"Create Property - Variable"`
- Omni Chest short description: `"Create Property - Crowdsale Issuance"`
- OmniJ function name: `"createCrowdsaleHex"` (let's ignore the "Hex" here)
- OmniJ part of a test label: `"create crowdsale"`
- OmniJ code comment: `"create a property via crowdsale with variable supply"`

Now this is more or less pretty similar, but when it comes to.. say for example DEx related stuff, there are `"trade tokens for bitcoins"`, `"sell bitcoins for mastercoins"`, `"create an offer"`, and other variations.

It would be nice to have a spreadsheet with:

| Type | Command          | Label              | Description                                            |
| ---- | ---------------- | ------------------ | ------------------------------------------------------ |
|  1   | simple_send      | "Simple Send"      | "Send tokens"                                          |
| 50   | create_crowdsale | "Create Crowdsale" | "Create a property via crowdsale with variable supply" |
| ...  | ...              | ...                | ...                                                    |

In particular, having consistent commands seems useful, for a consistent use of several Omni products. It's not relevant, whether it's `"simplesend"` or `"simple_send"`, but more of an issue, if one uses `"send"`, but the other `"transfer_tokens"`, especially when it comes to APIs. Hope you see my point? :)

More obvious cases: `"property"` or `"currency"`? `"tokens"` or `"coins"`, ...?

If I find some time, I'm going to create a list for Omni Core, which could then be passed around to collaboratively improve the names and labels.

I mentioned, I dislike `"send_MP"` and `"send_OMNI"`, but I also have no idea how to make it better - or what others think about this topic in general - so this can basically be considered as trying to find an answer and alternatives.. :)

At some point this can be extended to parameter names, and it would also be useful, if there were a compact list of the fields of RPC results (e.g. of `"gettransaction_MP"`).

### PR DESCRIPTION
**Pull request by dexX7**
This pull request resolves #14.

The descriptions are based on actual output, which I tested.

Please let me know, if there is something to change or refine.

### CODE REVIEW

**Comment by dexX7** [comment]
The complete output of all help descriptions:

- https://gist.githubusercontent.com/dexX7/d2aa23987aad2d351639/raw/53fd14d7e140482dc55ce9af2eec0523c63b8ff5/rpc_help

File: src/omnicore/rpc.cpp [hunk 1]
```diff
@@ -78,7 +78,7 @@ void MetaDexObjectToJSON(const CMPMetaDEx& obj, Object& metadex_obj)
     // add data to JSON object
     metadex_obj.push_back(Pair("address", obj.getAddr()));
     metadex_obj.push_back(Pair("txid", obj.getHash().GetHex()));
-    if (obj.getAction() == 4) metadex_obj.push_back(Pair("ecosystem", isTestEcosystemProperty(obj.getProperty()) ? "Test" : "Main"));
+    metadex_obj.push_back(Pair("ecosystem", isTestEcosystemProperty(obj.getProperty()) ? "test" : "main"));
     metadex_obj.push_back(Pair("propertyidforsale", (uint64_t) obj.getProperty()));
```
Comment (zathras-crypto) [inline src/omnicore/rpc.cpp]: Note, we don't provide an 'ecosystem' attribute in RPC output for any other call - the only reason it was included here was for cancel-by-ecosystem transactions to show what ecosystem the transaction applied to.

I don't really see the need to supply the ecosystem attribute except for that explicit case as it's an API and the ecosystem is easily devolved from already provided property ID data.  But mostly I don't like doing it just for one type of transaction only (MetaDEx trades) - we should either supply property ID only (preferred) or both property ID and ecosystem for all transactions.

File: src/omnicore/rpc.cpp [hunk 2]
```diff
@@ -763,28 +787,29 @@ Value getgrants_MP(const Array& params, bool fHelp)
     if (fHelp || params.size() != 1)
         throw runtime_error(
             "getgrants_MP propertyid\n"
-            "\nGet information about grants and revokes for a property identifier.\n"
+            "\nRetusn information about granted and revoked units of managed tokens.\n"
```
Comment (zathras-crypto) [inline src/omnicore/rpc.cpp]: Typo

**Comment by zathras-crypto** [comment]
Couple of comments, other than that looks good to me - do you want to merge and then I'll redo my RPC PR to resolve conflicts?

**Comment by dexX7** [comment]
Yup, sounds good. One moment to address the comments.

**Comment by zathras-crypto** [comment]
Thanks dude, I've resolved the conflicts but I'm getting a build failure here:
```
  CXX      omnicore/libbitcoin_server_a-notifications.o
In file included from omnicore/notifications.cpp:5:0:
./omnicore/version.h:44:37: error: ‘OMNICORE_VERSION_MILESTONE’ was not declared in this scope
                      +  100000000 * OMNICORE_VERSION_MILESTONE
                                     ^
./omnicore/version.h:45:37: error: ‘OMNICORE_VERSION_MAJOR’ was not declared in this scope
                      +     100000 * OMNICORE_VERSION_MAJOR
                                     ^
./omnicore/version.h:46:37: error: ‘OMNICORE_VERSION_MINOR’ was not declared in this scope
                      +        100 * OMNICORE_VERSION_MINOR
                                     ^
./omnicore/version.h:47:37: error: ‘OMNICORE_VERSION_PATCH’ was not declared in this scope
                      +          1 * OMNICORE_VERSION_PATCH;
                                     ^
Makefile:4338: recipe for target 'omnicore/libbitcoin_server_a-notifications.o' failed
```

**Comment by dexX7** [comment]
That's really strange. Did you change anything of your setup?

These `OMNICORE_VERSION_X` are defined just a few lines above:

 - https://github.com/OmniLayer/omnicore/blob/omnicore-0.0.10/src/omnicore/version.h

... or, if exists, via `src/config/bitcoin-config.h`.

Can you try to run `./configure` before building? I assume the config file exists, but is an outdated one from another branch.

**Comment by zathras-crypto** [comment]
> Can you try to run ./configure before building? I assume the config file exists, but is an outdated one from another branch.

Yeah I'd done that already mate - let me poke around a bit more

**Comment by zathras-crypto** [comment]
So inside version.h we have:
```
#if defined(HAVE_CONFIG_H)
#include "config/bitcoin-config.h"
#else
......define omni core version stuff...
#endif
```
which means the omni core version stuff is only defined in version.h if we don't have a bitcoin-config.h file.  I do have such a file, but the file does not contain any omni core version info.

I see what you're saying about it's perhaps an older version from another branch or something, but I've done a ```./configure``` and ```make clean``` already...

Hmm...

**Comment by zathras-crypto** [comment]
I'll try deleting the file manually, then ```./configure``` again.  FYI my configure line has nothing special in it (just ```./configure CPPFLAGS="-I${BDB_PREFIX}/include/ -O2" LDFLAGS="-L${BDB_PREFIX}/lib/" --with-gui```

**Comment by zathras-crypto** [comment]
No joy, deleted the file & re=ran ```./configure``` the file is recreated but it doesn't have any of the Omni Core version stuff (though it does have Omni Core package info).  The file that is generated is here pastie.org/pastes/10276781/text?key=mqbr95yzjuw1bsjnczgig

**Comment by zathras-crypto** [comment]
Think I've got it sorted.

Needed to re-run ```./autogen.sh``` before ```./configure``` - the file that was created after doing that does contain the Omni Core version defines.

Building now :)

**Pull request merged**